mlp/scripts/create_update_run.py

```diff
@@ -29,96 +29,118 @@ def check_environment_set():
         "OTHER_CLIENT_SECRET",
         "ENDPOINT",
         "NAMESPACE",
-        "MLP_VERSION",
-        "PROJECT_VERSION",
+        "MLP_VERSION"
     ]
     for environ in environs:
         if os.environ.get(environ) is None or os.environ.get(environ).strip() == "":
             sys.exit(
-                "{environ} not set. Must set to appropriate value before runnign this script.".format(
-                    environ=environ
-                )
+                "{environ} not set. Must set to appropriate value before runnign this script."
+                .format(environ=environ)
             )
 
 
-def create_update_run(pipeline_path: Text, experiment: Optional[Text] = "dev"):
+def create_update_run(
+        pipeline_path: Text, project_dir: Text, pipeline_name: Text,
+        pipeline_docker_path: Text, mlp_project: Text, experiment: Optional[Text] = "dev",
+        update: Optional[bool] = False, upgrade: Optional[bool] = False):
     check_environment_set()
-
-    # Pull in the pipeline file as a module to get various variables.
-    # pipeline_path = FLAGS.pipeline_path
-    pipeline_mod = pipeline_path.split("/")[-1].rstrip(".py")
-    spec = importlib.util.spec_from_file_location(pipeline_mod, pipeline_path)
-    mod = importlib.util.module_from_spec(spec)
-    spec.loader.exec_module(mod)
-
     # Create a unique identifier for the run.
     run_str = datetime.datetime.utcnow().strftime("%Y-%m-%d-%H-%M-%S")
 
-    # Get the standard variable names
-    var_names = pipeline_var_names(
-        mod._RUN_DIR,
-        run_str,
-        mod._MLP_PROJECT,
-        mod._MLP_SUBPROJECT,
-        mod._RUNNER,
-        mod._PIPELINE_TYPE,
-        experiment,
-    )
-    # Run the pipeline file to get the pipeline package to upload to kubeflow
-    run_pipeline_file(pipeline_path, run_str, experiment)
+    repo = Repo(search_parent_directories=True)
+    pattern = r'.*\.py$'  # regex pattern to match python files
+    exclude_pattern = r'.*(setup|__init__)\.py$'
+    changed_files = [item.a_path for item in repo.index.diff(None) + repo.index.diff('HEAD')]
+    changed_flag = False
+    for f in changed_files:
+        if re.match(pattern, f) and not re.match(exclude_pattern, f):
+            changed_flag = True
+            break
 
-    # Load the config file from gcp (generated from run_pipeline_file)
-    vc = VarConfig()
-    vc._load_from_json(var_names["vc_config_path"])
+    # Define the kubeflow client
+    client = kfp.Client(
+        host=os.environ.get("ENDPOINT"),
+        client_id=os.environ.get("CLIENT_ID"),
+        namespace=os.environ.get("NAMESPACE"),
+        other_client_id=os.environ.get("OTHER_CLIENT_ID"),
+        other_client_secret=os.environ.get("OTHER_CLIENT_SECRET"),
+    )
+    pipeline_id = client.get_pipeline_id(pipeline_name)
+    pipeline_versions = []
+    pipelines = client.list_pipeline_versions(pipeline_id, sort_by="created_at desc").versions
+    if pipeline_id and pipelines:
```

Comment:
This is getting a little too long, it's probably better to package up and put into a separate function if possible.

Reply:
please check the new version

Reply:
Sorry I meant the block of lines from 75 - 93